libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -239,28 +241,44 @@ void AP_Proximity_RPLidarA2::send_request_for_health()
     uint8_t tx_buffer[2] = {RPLIDAR_PREAMBLE, RPLIDAR_CMD_GET_DEVICE_HEALTH};
     _uart->write(tx_buffer, 2);
     _last_request_ms = AP_HAL::millis();
+    Debug(1, "LIDAR HEALTH REQUEST ACTIVATED");
     _rp_state = rp_health;
 }
 
+// send request for sensor sampling rate
+void AP_Proximity_RPLidarA2::send_request_for_rate()                                    //not called yet
+{
+    if (_uart == nullptr) {
+        return;
+    }
+    uint8_t tx_buffer[2] = {RPLIDAR_PREAMBLE, RPLIDAR_CMD_GET_DEVICE_RATE};
+    _uart->write(tx_buffer, 2);
+    _last_request_ms = AP_HAL::millis();
+    Debug(1, "LIDAR RATE REQUEST ACTIVATED");
+    _rp_state = rp_rate;
+}
+
+
 void AP_Proximity_RPLidarA2::get_readings()
 {
     if (_uart == nullptr) {
         return;
     }
-    Debug(2, "             CURRENT STATE: %d ", _rp_state);
+
+    Debug(3, "             CURRENT STATE: %d ", _rp_state);
     uint32_t nbytes = _uart->available();
 
     while (nbytes-- > 0) {
 
         uint8_t c = _uart->read();
-        Debug(2, "UART READ %x <%c>", c, c); //show HEX values
+        Debug(3, "UART READ %x <%c>", c, c); //show HEX values
 
         STATE:
         switch(_rp_state){
 
             case rp_resetted:
                 Debug(3, "                  BYTE_COUNT %d", _byte_count);
-                if ((c == 0x52 || _information_data) && _byte_count < 62) {
+                if ((c == 0x52 || _information_data) && _byte_count < 55/*62*/) {
```

Comment:
Please remove the commented out code

Reply:
That's some intermediate commit, no such line in the final revision at all

Reply:
A3M1 is 64 bytes.

52 50 20 4c 49 44 41 52 20 53 79 73 74 65 6d 2e 
0d 0a 46 69 72 6d 77 61 72 65 20 56 65 72 20 31 
2e 32 36 20 2d 20 72 63 31 2c 20 48 57 20 56 65 
72 20 36 0d 0a 4d 6f 64 65 6c 3a 20 33 31 0d 0a

Reply:
That's interesting, yet another message length for another model! That was the reason not to count the bytes anymore (as implemented in final revision).